client/modules/IDE/components/Header/MobileNav.jsx

```diff
@@ -0,0 +1,338 @@
+import React, { useContext, useMemo } from 'react';
+import styled from 'styled-components';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { useLocation } from 'react-router';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import { prop, remSize } from '../../../../theme';
+import AsteriskIcon from '../../../../images/p5-asterisk.svg';
+import IconButton from '../../../../components/mobile/IconButton';
+import { AccountIcon, EditorIcon, MoreIcon } from '../../../../common/icons';
+import {
+  newFile,
+  newFolder,
+  openPreferences,
+  showKeyboardShortcutModal
+} from '../../actions/ide';
+import { logoutUser } from '../../../User/actions';
+import { useSketchActions } from '../../hooks';
+import { CmControllerContext } from '../../pages/IDEView';
+import { selectSketchPath } from '../../selectors/project';
+
+const Nav = styled.div`
+  background: ${prop('MobilePanel.default.background')};
+  color: ${prop('primaryTextColor')};
+  padding: ${remSize(8)} 0;
+  gap: ${remSize(14)};
+  display: flex;
+  align-items: center;
+  font-size: ${remSize(10)};
+  box-shadow: #00000030 0px 2px 8px 0px;
+  z-index: 10;
+`;
+
+const LogoContainer = styled.div`
+  width: ${remSize(28)};
+  aspect-ratio: 1;
+  margin-left: ${remSize(14)};
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  > svg {
+    width: 100%;
+    height: 100%;
+    > path {
+      fill: ${prop('accentColor')};
+      stroke: ${prop('accentColor')};
+    }
+  }
+`;
+
+const Title = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${remSize(2)};
+
+  > * {
+    padding: 0;
+    margin: 0;
+  }
+
+  > h5 {
+    font-size: ${remSize(13)};
+    font-weight: normal;
+  }
+`;
+
+const Options = styled.div`
+  margin-left: auto;
+  display: flex;
+  /* transform: translateX(${remSize(12)}); */
+  svg {
+    fill: ${(props) => props.color};
+  }
+
+  /* Drop Down menu */
+  > div > button:focus + ul,
+  > div > ul > button:focus ~ div > ul {
```

Comment:
It seems like you are using CSS-only to open and close the menus, which is very cool, but it might get in the way of implementing the behavior that we discussed where clicking on the * icon would also open the menu.

It also has impacts on accessibility, which, I don't really know what to make of that when we're dealing with mobile.  But a keyboard user won't be able to tab through the menu items.

So we probably need some sort of React state and then use the state to control an `opened` class on the menu list.  I need to think about how we can reuse the focus/blur logic from the `NavBar` here.

Reply:
I'm playing with using all the contexts and states from the `NavBar` and I've got it like 90% working but I'm tired and don't want to deal with Git stuff now so I'll send it to you tomorrow.

Reply:
when I implemented it I thought most of the users won't use the keyboard on a mobile device and implemented it like this, but what you did is great and we should definitely include that